src/resolvers/Mutation/removeComment.ts

```diff
@@ -48,38 +52,43 @@ export const removeComment: MutationResolvers["removeComment"] = async (
     );
   }
 
-  // Checks whether currentUser with _id === context.userId is not the creator of comment.
-  if (comment.creator.toString() !== context.userId.toString()) {
+  const isCurrentUserAdminOfOrganization = currentUser.adminFor.some(
+    (organization) =>
+      organization.toString() === comment.postId.organization.toString()
+  );
+
+  // Checks whether currentUser with _id === context.userId has the authorization to delete the comment
+  if (
+    currentUser.userType !== "SUPERADMIN" &&
+    !isCurrentUserAdminOfOrganization &&
+    comment.creator.toString() !== context.userId.toString()
+  ) {
     throw new errors.UnauthorizedError(
       requestContext.translate(USER_NOT_AUTHORIZED_ERROR.MESSAGE),
       USER_NOT_AUTHORIZED_ERROR.CODE,
       USER_NOT_FOUND_ERROR.PARAM
     );
   }
 
-  /*
-  Removes comment._id from comments list and reduces commentCount
-  by 1 of post with _id === comment.post
-  */
+  // Reduce the commentCount by 1 of the post with _id === commentPost.postId
   await Post.updateOne(
     {
-      _id: comment.post,
+      _id: comment!.postId,
     },
     {
-      $pull: {
-        comments: comment._id,
-      },
       $inc: {
         commentCount: -1,
       },
     }
   );
 
-  // Deletes the comment.
+  // Deletes the comment
   await Comment.deleteOne({
-    _id: comment._id,
+    _id: args.id,
```

Comment:
You should stick to a convention for this. Either use `args.id` or use the database object you fetched `object.id`. Makes the code cleaner.